inspect.go

```diff
@@ -31,7 +31,7 @@ type InspectOptions struct {
 }
 
 // Inspect makes an admin call to download a raw files from disk.
-func (adm *AdminClient) Inspect(ctx context.Context, d InspectOptions) (key [32]byte, c io.ReadCloser, err error) {
+func (adm *AdminClient) Inspect(ctx context.Context, d InspectOptions) (format byte, key []byte, c io.ReadCloser, err error) {
```

Comment:
@vadmeste Let's return something called as `InspectReader`  that automatically 

- decrypts if key is present
- detects which format the inspect data is

We have no reason to expose this to the caller - caller should simply `io.ReadAll(ird)`  IMO.

Reply:
@harshavardhana we actually need to return a zip file which contains a plaintext file and some encrypted data. The zip file will be sent by email or uploaded in a portal :)

The zip file needs to be stored in a disk first, because decompressing it requires a seek feature io.ReaderAt

Now, why the server generates zip file in the first place? because the size of files inside it are stored after the data itself and this is needed so we can feed the zip file in a streaming fashion.

So overall InspectReader to decrypt on the fly is not possible.

Reply:
Again all of that is possible using a single reader @vadmeste currently functionality of replying key and all barebones is complicated.

The logic of extracting plaintext, and encrypted must be in this SDK alone - caller implementations can diverge and can lead to bugs. 

You return something called as `*inspectReader` now this reader if you do `io.Copy()` gives the encrypted content however if you do `(ir *InspectReader) ClusterInfo() Info` returns the relevant info.

Reply:
> Again all of that is possible using a single reader @vadmeste 

This is only possible if we save the data in the disk, which is probably not very elegant as an SDK API.

I think let's do another round of discussion, maybe we don't need all of this.

Reply:
How you can keep them in memory @vadmeste ?

Reply:
> How? - you can keep them in memory @vadmeste ?

Keep the whole inspect data in the memory ? the inspect data can be huge

Reply:
Not whole just the one that unencrypted.

Reply:
The current implementation is very low level we have to fix it because this looks quite strange implementation wise.

As it expects mc to do all the work by returning the entire skeleton of the protocol.

We should never have callers do so much work IMO, we have to come up with a simple API that returns reader.

This reader may have additional things so it can be a InspectReader that allows manipulating the stream but never the way it is implemented in the PR.

We are expecting callers to have their own implementations of interpreting the returned values here. 

This leads to diverged implementations and future breakage.